src/engine.rs

```diff
@@ -20,6 +20,16 @@ pub struct ProfileOpts {
 #[folder = "providers/"]
 struct StandardProviders;
 
+fn uses_msgpack_provider(module: &Module) -> bool {
+    let imported_modules: HashSet<String> =
+        module.imports().map(|i| i.module().to_string()).collect();
+
+    imported_modules.iter().any(|name| {
+        name.starts_with("shopify_function_v")
+            || (name.starts_with("shopify_functions_javy_v") && name != "shopify_functions_javy_v1")
```

Comment:
Do you mind comparing against `v2` directly? I think it's probably safer to ensure that we're comparing against the exact version rather than a wildcard

Reply:
updated, originally went with same approach as we were using for the wasm api, but makes more sense to hard code it here for now since v1 doesn't use.